### PR TITLE
add "how it works" and "tools" sections

### PR DESCRIPTION
**Pull request by raphaelreyna**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,6 +308,24 @@ The telepresence background processes `daemon` and `connector` both produces log
 - macOS `~/Library/Logs/telepresence`
 - Linux `~/.cache/telepresence/logs`
 
+## How it works
+
+When Telepresence 2 connects to a Kubernetes cluster, it
+
+ 1. Ensures Traffic Manager is installed in the cluster.
+ 2. Looks for the relevant subnets in the kubernetes cluster.
+ 3. Creates a virtual network device called a _TUN device_ and names it `tel0`.
```
Comment (thallgren) [inline README.md:317]: The device is named `tun0` on macOS (or `tun1`, `tun2`, etc. whichever is available) so using "tel0" in the text can be confusing for a Mac user.

File: README.md [hunk 2]
```diff
@@ -308,6 +308,24 @@ The telepresence background processes `daemon` and `connector` both produces log
...
+ 4. Assigns a non-conflicting IP address to `tel0` for each of the subnets.
+ 5. Binds itself to `tel0` and starts listening for traffic from `tel0` and Traffic Manager.
+ 6. Starts listening for and serving DNS requests.
+
+When a locally running application makes a network request to a service in the cluster, Telepresence will resolve the name to an address within the cluster.
+The operating system then sees that the TUN device has an address in the same subnet as the address of the outgoing packets and sends them to `tel0`.
+Telepresence is on the other side of `tel0` and picks up the packets, injecting them into the cluster through a gRPC connection with Traffic Manager.
```
Comment (thallgren) [inline README.md:324]: The packets aren't really "injected". The data of the packages are tunneled over a stream (in packets that are much larger than the TCP packets). I suggest you use different wording here unless you want to go into detail on how that works. How about "The operating system then sees that the address is covered by one of the TUN device's subnets, which means that telepresence will route the traffic to the cluster through a gRPC connection with the Traffic Manager"?

**Review by thallgren — CHANGES_REQUESTED** [review]

**Comment by thallgren** [comment]
@raphaelreyna can you please rebase and update this one so that we can merge it?

**Comment by DanielYWoo** [comment]
could you also put some explanation about the DNS? How the domain names are resolved by coreDNS outside of the cluster?

**Pull request merged**